patches/server/0738-Hide-unnecessary-itemmeta-from-clients.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Tue, 3 Aug 2021 17:28:27 +0200
+Subject: [PATCH] Hide unnecessary itemmeta from clients.
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index c334f29c69c1e6e3fe55cd6695e7df400cf36058..edfd20abcabc9c2046fa419799ce089ddf2e0e5b 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -863,6 +863,11 @@ public class PaperWorldConfig {
+         behaviorTickRates = loadTickRates("behavior");
+     }
+ 
++    public boolean hideItemmetaFromClients = false;
++    private void getHideItemmetaFromClients() {
++        hideItemmetaFromClients = getBoolean("hide-itemmeta-from-clients", hideItemmetaFromClients);
++    }
++
+     private Table<String, String, Integer> loadTickRates(String type) {
+         log("  " + type + ":");
+         Table<String, String, Integer> table = HashBasedTable.create();
+diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+index 2b7eeb5659b1083ef550eb9feb0b7ba8a92a92e3..4025f156222ff09f3cb6d8c42aef5d9e0a0d845f 100644
+--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
++++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+@@ -3041,6 +3041,53 @@ public abstract class LivingEntity extends Entity {
+         equipment.forEach((enumitemslot, itemstack) -> {
+             ItemStack itemstack1 = itemstack.copy();
+ 
++            // Paper start - hide unnecessary itemmeta from clients.
++            if (itemstack1.getTag() != null && this.level.paperConfig.hideItemmetaFromClients) {
++                CompoundTag nbt = itemstack1.getTag();
++                itemstack1.setCount(1);
```

Comment:
We should preserve showing plural for resource packs using OptiFine's stuff. Just doing e.g. `itemstack1.setCount(itemstack1.getCount() > 1 ? 2 : 1)` is enough.